### PR TITLE
docs: add documentation to state differences from the git cli

### PR DESCRIPTION
**Pull request by cjhoward92**
Started potential documentation to list differences in functionality from the of the Git CLI. If anyone has other things they would like to include, such as missing functionality, or if they would like wording changed please let me know.

This is in response to issue [#4583](https://github.com/libgit2/libgit2/issues/4583)

### CODE REVIEW

**Comment by pks-t** [comment]
Thanks for this PR! It was one of our takeaways from the recent lbgit2 summit to start such documentation, so this is definitely welcome. We could also extend this document for features not (yet) implemented in libgit2 later

**Comment by cjhoward92** [comment]
I'm glad you like the PR! Is there anything you would have me add now, or anything related to the structure of the document that you would like changed?

File: docs/differences-from-git-cli.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Differences from the Git CLI
+
+In some instances, the functionality of libgit2 deviates slightly from that of the Git CLI. This can because of technical limitations when developing a library, licensing limitations when converting functionality from the CLI to libgit2, or various other reasons.
+
+Repository and Workdir Path Reporting
+-------------------------------------
+
+When retrieving the absolute path of a repository from the Git CLI, one could expect the output to lool like so:
```
[the remark this answers is not in the document]
Comment (cjhoward92) [reply docs/differences-from-git-cli.md]: Nice, this is much more concise. I didn't know how much detail I should add so I was winging it. I like this better.

**Comment by cjhoward92** [comment]
I want to add more to this document but it's been a while since I was in `libgit2` so I am blanking on some other differences. If I think of any more before you merge this then I will update it! Or if you have any suggestions!

**Comment by pks-t** [comment]
It's a good start, so I just went ahead and merged it. We can (and probably should) just amend this document as time goes by. Thanks!

**Pull request merged**